### ISSUE TITLE
Manually set basepath for nextra 

### ISSUE THREAD

**Issue by Kamasah-Dickson** [shuding/nextra#2720]
I am using next js 14 app route and I have the /pages folder where my nextra docs is located. I want my nextra documentation to be in the /docs so you need to got to /docs to see the documentation. But when I do that /docs is being included in the sidebar of the nextra docs because nextra looks at the base path / it will be better if we can manually set a base path.

**Comment by mutasim77** [comment]
You can do this by using [middlewares](https://nextjs.org/docs/app/building-your-application/routing/middleware#conditional-statements) from Next.js, essentially by rewriting paths.

**Comment by Kamasah-Dickson** [comment]
Didn't thought about that but I will try that out and see

**Comment by Kamasah-Dickson** [comment]
> You can do this by using [middlewares](https://nextjs.org/docs/app/building-your-application/routing/middleware#conditional-statements) from Next.js, essentially by rewriting paths.

Didn't thought about that but I will try that out and see

**Comment by Kamasah-Dickson** [comment]
> You can do this by using [middlewares](https://nextjs.org/docs/app/building-your-application/routing/middleware#conditional-statements) from Next.js, essentially by rewriting paths.

It looks promising but considering the fact that am already utilizing the / in the app route which means  in the /pages dir I no longer have access to the / it's either /something. Now i have to navigate to like /get-started or one of the mdx pages to see the documentation. Imaging this many mdx paths how am I gonna do it because changing each path manually is  not an  easy task. I hope you understand what am trying to say

**Comment by mutasim77** [comment]
May I know what you want to build and your folder structure?(If it's not a big problem).  Based on that, maybe I can help you.

**Comment by Kamasah-Dickson** [comment]
> May I know what you want to build and your folder structure?(If it's not a big problem).  Based on that, maybe I can help you.

I am building a saas and am using the app route. I want to protect the  documentation page using next js middleware and based on the subscription plan, they can access the docs but Unfortunately nextra only works in the /pages dir for now and am lucky in next 14 the pages and app dir can be used together which is good news. Unfortunately nextra is looking at the root of the /pages and includes all the folders in its sidebar so having something like **/docs/all the mdx files**, docs wraps around the files and is included as a parent folder in the sidebar which is not what I want. I want to visit /docs to see the documentation without /docs wrapping around all the mdx file as a parent folder.

**Comment by Kamasah-Dickson** [comment]
you can take a look at this mine is like this. they have both the /app and the /pages and in some way they managed to put the documentation in the /docs.

https://github.com/chakra-ui/panda/tree/main/website

**Comment by Kamasah-Dickson** [comment]
> you can take a look at this mine is like this. they have both the /app and the /pages and in some way they managed to put the documentation in the /docs.
> 
> https://github.com/chakra-ui/panda/tree/main/website

after looking at their folder structure I was able to find the solution.

- I just created a folder satsat-ai-docs as place all the nextra docs inside, acting as the root folder.
- I added _meta.json to the root and described the type to be a page. Setting it as a page will appear as a nav link in the navbar, in my case i dont need it as a nav link so i hide it by setting the display to hidden

![Capture](https://github.com/shuding/nextra/assets/86136379/5dad802f-55f1-4083-8d31-886f904a475c)
